population of related model gives - Option "ref" for "Game.topic" is null/undefined

## Versions

```
npm ls mongoose
bot@1.0.0 /Users/dc/dev/ten/puzzleparty/server
└── mongoose@5.10.18

npm ls @typegoose/typegoose
bot@1.0.0 /Users/dc/dev/ten/puzzleparty/server
└── @typegoose/typegoose@7.4.4

npm -v
6.14.9
node -v
v14.5.0
npx tsc -v
Version 4.1.2
```

## What is the Problem?
In using mongoose basic populate methods to hydrate a linked model:

https://typegoose.github.io/typegoose/docs/api/virtuals/#virtual-populate

I am getting a "ref" is undefined for trying to link to a related Class.

Error: Option "ref" for "Game.topic" is null/undefined! [E005]

## Code Example

I have game and topic classes I am trying to 'populate' as below.
So I could use `game.topic` in a result set.
But at compile time I get an error with this.

```ts

class Topic {

  @prop({
    ref: () => Game,
    foreignField: 'topic',
    localField: '_id',
    justOne: true
  })
  public game: Ref<Game>

 ...
}

class Game {
  @prop({ ref: () => Topic })
  public topic: Ref<Topic>
 ...
}
```

The full error is:

```
Error: Option "ref" for "Game.topic" is null/undefined! [E005]
    at Object.processProp (/Users/dc/dev/ten/puzzleparty/server/node_modules/@typegoose/typegoose/src/internal/processProp.ts:141:7)
    at Object._buildSchema (/Users/dc/dev/ten/puzzleparty/server/node_modules/@typegoose/typegoose/src/internal/schema.ts:58:7)
    at buildSchema (/Users/dc/dev/ten/puzzleparty/server/node_modules/@typegoose/typegoose/src/typegoose.ts:140:9)
    at Object.getModelForClass (/Users/dc/dev/ten/puzzleparty/server/node_modules/@typegoose/typegoose/src/typegoose.ts:84:58)
    at Object.<anonymous> (/Users/dc/dev/ten/puzzleparty/server/src/models/Game.ts:93:19)
    at Module._compile (internal/modules/cjs/loader.js:1201:30)
    at Module._compile (/Users/dc/dev/ten/puzzleparty/server/node_modules/source-map-support/source-map-support.js:547:25)
    at Module.m._compile (/private/var/folders/wl/fsxhspwn0351_lx_55rkwgwc0000gn/T/ts-node-dev-hook-467125897881844.js:57:33)
    at Module._extensions..js (internal/modules/cjs/loader.js:1221:10)
    at require.extensions.<computed> (/private/var/folders/wl/fsxhspwn0351_lx_55rkwgwc0000gn/T/ts-node-dev-hook-467125897881844.js:59:20)
    at Object.nodeDevHook [as .ts] (/Users/dc/dev/ten/puzzleparty/server/node_modules/ts-node-dev/lib/hook.js:63:13)
    at Module.load (internal/modules/cjs/loader.js:1050:32)
    at Function.Module._load (internal/modules/cjs/loader.js:938:14)
    at Module.require (internal/modules/cjs/loader.js:1090:19)
    at require (internal/modules/cjs/helpers.js:75:18)
    at Object.<anonymous> (/Users/dc/dev/ten/puzzleparty/server/src/models/Topic.ts:12:1)
```


## Do you know *why* it happens?

*no*
Only thought was that if there is a circular dependency between the types since I have to import one class into the other.

I am getting the same error myself.
I am using typegoose associated with type-graphql.

**User Model**
```
@ObjectType({ description: 'The Users model' })
export class User {
  @Field(() => ObjectIdScalar)
  readonly _id: ObjectId;

  @Field(() => String, { description: 'User name' })
  @prop({ trim: true, required: true })
  name: string;

  @Field({ description: 'User surname', nullable: true })
  @prop({ trim: true })
  surname?: string;

  @Field({ description: 'User email to be used on login' })
  @prop({ trim: true, required: true, unique: true })
  email: string;

  @prop()
  password?: string;

  @Field(_type => [Post])
  @prop({ type: () => Post, default: [] })
  posts: Post[];

  @Field({ description: 'User creation date' })
  createdAt?: Date;
  
  @Field({ description: 'User last update date', nullable: true })
  updatedAt?: Date;

  static async encryptPassword(password: string): Promise<string> {
    return hash(password);
  }

  public async authenticate(plainText: string) {
    try {
      return verify(this.password || '', plainText);
    } catch (err) {
      return false;
    }
  }
}

export const UserModel = getModelForClass(User, { schemaOptions: {
  collection: 'users',
  timestamps: {
    createdAt: 'createdAt',
    updatedAt: 'createdAt',
  },
} });
```

**Post Model**
```
@ObjectType({ description: 'The Posts model' })
export class Post {
  @Field(() => ObjectIdScalar)
  readonly _id: ObjectId;

  @Field(() => String, { description: 'Post title' })
  @prop({ trim: true, required: true })
  title: string;

  @Field({ description: 'Post content' })
  @prop({ required: true })
  content: string;

  @Field({ description: 'Number of likes this post has received', defaultValue: 0 })
  @prop({ default: 0 })
  likes?: number;

  @Field(_type => User)
  @prop({ ref: User, required: true })
  creator: Ref<User>;

  @Field({ description: 'Post creation date' })
  createdAt?: Date;
  
  @Field({ description: 'Post last update date', nullable: true })
  updatedAt?: Date;
}

export const PostModel = getModelForClass(Post, { schemaOptions: {
  collection: 'posts',
  timestamps: {
    createdAt: 'createdAt',
    updatedAt: 'createdAt',
  },
} });
```

**This is the error I'm getting.**
```
Error: Option "ref" for "Post.creator" is null/undefined! [E005]
    at Object.processProp (/Users/brunocruz/code/creators-communities-project/server/node_modules/@typegoose/typegoose/src/internal/processProp.ts:141:7)
    at Object._buildSchema (/Users/brunocruz/code/creators-communities-project/server/node_modules/@typegoose/typegoose/src/internal/schema.ts:58:7)
    at buildSchema (/Users/brunocruz/code/creators-communities-project/server/node_modules/@typegoose/typegoose/src/typegoose.ts:140:9)
    at Object.getModelForClass (/Users/brunocruz/code/creators-communities-project/server/node_modules/@typegoose/typegoose/src/typegoose.ts:84:58)
    at Object.<anonymous> (/Users/brunocruz/code/creators-communities-project/server/src/models/Post.ts:48:26)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Module._compile (/Users/brunocruz/code/creators-communities-project/server/node_modules/source-map-support/source-map-support.js:547:25)
    at Module.m._compile (/private/var/folders/s8/1hc7w_9577qc8ltsl4w7vd5h0000gn/T/ts-node-dev-hook-1372691903688159.js:57:33)
    at Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at require.extensions.<computed> (/private/var/folders/s8/1hc7w_9577qc8ltsl4w7vd5h0000gn/T/ts-node-dev-hook-1372691903688159.js:59:20)
    at Object.nodeDevHook [as .ts] (/Users/brunocruz/code/creators-communities-project/server/node_modules/ts-node-dev/lib/hook.js:63:13)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Module.require (internal/modules/cjs/loader.js:848:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/Users/brunocruz/code/creators-communities-project/server/src/models/User.ts:5:1)
[ERROR] 18:24:43 Error: Option "ref" for "Post.creator" is null/undefined! [E005]
```

Any ideas of whats should I do to correct this error?

--- a/docs/api/virtuals.md
+++ b/docs/api/virtuals.md
@@ -58,7 +58,7 @@ Virtual-Populate is also supported by Typegoose
 
 Options ([look here for more details](https://mongoosejs.com/docs/api/schema.html#schema_Schema-virtual)):
 
-- `ref`: This is like a normal ref **[Required]**
+- `ref`: This is like a normal [ref](https://typegoose.github.io/typegoose/docs/api/decorators/prop/#ref), use `'ClassName'` when the classes are in different files **[Required]**
 - `foreignField`: Which property(on the ref-Class) to match `localField` against **[Required]**
 - `localField`: Which property(on the current-Class) to match `foreignField` against **[Required]**
 - `justOne`: Return as One Document(true) or as Array(false) ***[Optional]***

